test/README.md

```diff
@@ -0,0 +1,92 @@
+# Test
+
+This directory contains tests and testing docs for `Knative Eventing`.
+
+* [Unit tests](#running-unit-tests) reside in the codebase alongside the code they test
+* [End-to-end tests](#running-end-to-end-tests) reside in [`/test/e2e`](./e2e)
+
+## Running unit tests
+
+Use `go test`:
+
+```shell
+go test -v ./pkg/...
+```
+
+_By default `go test` will not run [the e2e tests](#running-end-to-end-tests), which need [`-tags=e2e`](#running-end-to-end-tests) to be enabled._
+
+## Presubmit tests
+
+[`presubmit-tests.sh`](./presubmit-tests.sh) is the entry point for the [end-to-end tests](/test/e2e).
+
+This script, and consequently, the e2e tests will be run before every code submission. You can run these tests manually with:
+
+```shell
+test/presubmit-tests.sh
+```
+
+_Note that to run `presubmit-tests.sh` or `e2e-tests.sh` scripts, you need to have a running environment that meets
+[the e2e test environment requirements](#environment-requirements)_
+
+## Running end-to-end tests
+
+To run [the e2e tests](./e2e), you need to have a running environment that meets
+[the e2e test environment requirements](#environment-requirements), and you need to specify the build tag `e2e`.
+
+```bash
+go test -v -tags=e2e -count=1 ./test/e2e
+```
+
+### One test case
+
+To run one e2e test case, e.g. TestKubernetesEvents, use [the `-run` flag with `go test`](https://golang.org/cmd/go/#hdr-Testing_flags):
+
+```bash
+go test -v -tags=e2e -count=1 ./test/e2e -run ^TestKubernetesEvents$
+```
+
+### Environment requirements
+
+There's couple of things you need to install before running e2e tests locally.
+
+1. `kubetest` installed:
+    ```bash
+    go get -u k8s.io/test-infra/kubetest
+    ```
+2. [A running `Knative Serving` cluster.]
+3. A docker repo containing [the test images](#test-images)
+
+Simply run the `./test/e2e-tests.sh` script. It will create a GKE cluster, install Knative Serving stack with Istio, upload test images to your Docker repo and run the end-to-end tests against the Knative Eventing built from source.
```

Comment:
```suggestion
Simply run the `./test/e2e-tests.sh` script. It will create a GKE cluster, install Knative Serving stack with Istio, upload test images to your Docker repo and run the end-to-end tests against the Knative Eventing Sources built from source.
```